controllers/baseboardmanagement_controller.go

```diff
@@ -107,7 +110,7 @@ func (r *BaseboardManagementReconciler) Reconcile(ctx context.Context, req ctrl.
 	return r.reconcile(ctx, baseboardManagement, baseboardManagementPatch, logger)
 }
 
-func (r *BaseboardManagementReconciler) reconcile(ctx context.Context, bm *bmcv1alpha1.BaseboardManagement, bmPatch client.Patch, logger logr.Logger) (ctrl.Result, error) {
+func (r *BaseboardManagementReconciler) reconcile(ctx context.Context, bm *bmcv1alpha1.BaseboardManagement, bmPatch client.Patch, logger logr.Logger) (_ ctrl.Result, reterr error) {
```

Comment:
I actually did this to aggregate the individual field reconciler errors and return a final one [147](https://github.com/tinkerbell/rufio/pull/5/files#diff-6ea8f2e85d97c6f1f0f4cd6f5905f72792e2b7db0052a83c663a8d31d61aa498R147).

We could have a global error instead?

Reply:
We don't want a global error; avoid globals (I can elaborate offline).

We can just define the error variable instead of letting the return parameter declare the variable. This avoids confusion and accidental behavior. In general, avoid named returns. They're only really necessary when you return 2 values of the same type and even then you might consider adjusting your function first.

Reply:
My bad. I just read what I wrote again 🤦 
I actually in my head meant a variable in the function scope that has the aggregated errors.
And removed the named returns 👍